catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -539,4 +586,38 @@ public FilterBuilder getFilterBuilder() {
         return filterBuilder;
     }
 
+    /**
+     * Copy the attributes from a metacard to another metacard.
+     *
+     * @param sourceMetacard      the source metacard
+     * @param metacardType        copy all attributes represented by this metacard type
+     * @param destinationMetacard the destination metacard
+     */
+    public void copyAttributes(Metacard sourceMetacard, MetacardType metacardType,
+            Metacard destinationMetacard) {
+        metacardType.getAttributeDescriptors()
+                .forEach(attributeDescriptor -> Optional.ofNullable(sourceMetacard.getAttribute(
+                        attributeDescriptor.getName()))
```

Comment:
✏️ Not critical, but If we know that `sourceMetacard` isn't null, we could make this
```java
Optional.of(attributeDescriptor.getName())
    .map(sourceMetacard::getAttribute)
```